### PR TITLE
Add information on the various package states

### LINKED ISSUE #35881: Ansible docs don't explain what "present" , "installed" etc mean
##### ISSUE TYPE

 - Documentation Report

##### COMPONENT NAME
yum module

webdocs - http://docs.ansible.com/ansible/latest/yum_module.html

##### SUMMARY
The meaning and functionality of "present"  "installed" and other states are either not mentioned in the docs, or is hard to locate. Answers to questions in stackoverflow reveal the same https://stackoverflow.com/questions/40410270/what-is-the-difference-between-two-state-option-values-present-and-install

##### STEPS TO REPRODUCE
view http://docs.ansible.com/ansible/latest/yum_module.html

##### EXPECTED RESULTS
People should gather information about what the parameters mean from the docs
##### ACTUAL RESULTS
People resort to blog posts and stack overflow

### PR DESCRIPTION
**Pull request by anchat1990**
##### SUMMARY
Added details on functionality of the available states in the yum module in documentation.
Fixes #35881

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
yum module

##### ANSIBLE VERSION
latest

### CODE REVIEW

File: lib/ansible/modules/packaging/os/yum.py [hunk 1]
```diff
@@ -43,6 +43,9 @@
   state:
     description:
       - Whether to install (C(present) or C(installed), C(latest)), or remove (C(absent) or C(removed)) a package.
+      - C(present) and C(present) will simply ensure that a desired package is installed.
```
Comment (Akasurde) [inline lib/ansible/modules/packaging/os/yum.py]: Possible typo.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/packaging/os/yum.py [hunk 2]
```diff
@@ -43,6 +43,9 @@
   state:
     description:
       - Whether to install (C(present) or C(installed), C(latest)), or remove (C(absent) or C(removed)) a package.
+      - C(present) and C(installed) will simply ensure that a desired package is installed.
+      - C(latest) will update the specified package if it’s not of the latest available version.
```
Comment (Akasurde) [inline lib/ansible/modules/packaging/os/yum.py]: s/it’s/its's/

Special character.

**Comment by anchat1990** [comment]
Hi @Akasurde , wondering if this is looking better now? Or am I missing something 😄

**Comment by Akasurde** [comment]
I am OK with changes, need an additional pair of eyes.

cc @mkrizek

File: lib/ansible/modules/packaging/os/yum.py [hunk 3]
```diff
@@ -43,6 +43,9 @@
   state:
     description:
       - Whether to install (C(present) or C(installed), C(latest)), or remove (C(absent) or C(removed)) a package.
```
Comment (mkrizek) [inline lib/ansible/modules/packaging/os/yum.py:45]: How about just this fix?
```
- Whether to install (C(present) or C(installed)), update (C(latest)) or remove (C(absent) or C(removed)) a package.
```
It seems clear enough for me. However, just a suggestion, I am fine with merging this PR.

**Review by Akasurde — APPROVED** [review]

**Pull request merged**